internal/credential/state.go

```diff
@@ -222,59 +163,58 @@ func (s *AwsCredentialPersistedState) DeleteCreds() error {
 		return err
 	}
 
-	s.AccessKeyId = ""
-	s.SecretAccessKey = ""
+	s.CredentialsConfig = nil
 	s.CredsLastRotatedTime = time.Time{}
 	return nil
 }
 
 // GetSession returns a configured AWS session for the credentials in
 // the state.
 func (s *AwsCredentialPersistedState) GetSession() (*session.Session, error) {
-	c, err := awsutil.NewCredentialsConfig(append([]awsutil.Option{
-		awsutil.WithAccessKey(s.AccessKeyId),
-		awsutil.WithSecretKey(s.SecretAccessKey),
-		awsutil.WithRegion(s.region),
-	}, s.testOpts...)...)
-	if err != nil {
-		return nil, err
-	}
-
-	return c.GetSession()
+	return s.CredentialsConfig.GetSession(s.testOpts...)
 }
 
-// ToMap returns a map of the credentials stored in the persisted state,
-// which includes the following keys: access_key_id, secret_access_key,
-// & creds_last_rotated_time
+// ToMap returns a map of the credentials stored in the persisted state.
+// ToMap will return an empty map for temporary credentials. ToMap will
+// return a map for long-term credentials with following keys:
+// access_key_id, secret_access_key & creds_last_rotated_time
 func (s *AwsCredentialPersistedState) ToMap() map[string]any {
+	if HasDynamicCredentials(s.CredentialsConfig.AccessKey) {
```

Comment:
Any reason to not have `HasDynamicCredentials` be a method of `CredentialsConfig` that gets the value of the accesskey from the struct and checks it instead of the caller passing the string in?